nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,23 +182,83 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
   }
 
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    getRef().checkMutable();
+
+    if (namespace.isEmpty()) {
+      throw new IllegalArgumentException("Creating empty namespaces is not supported");
+    }
+
+    ContentKey key = ContentKey.of(namespace.levels());
+    org.projectnessie.model.Namespace content =
+        org.projectnessie.model.Namespace.of(key.getElements(), metadata);
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
-      refresh();
-    } catch (NessieNamespaceAlreadyExistsException e) {
-      throw new AlreadyExistsException(e, "Namespace already exists: %s", namespace);
+
+      // First, check that the namespace doesn't exist at the current hash;
+      // this will avoid a NessieBadRequestException when committing the namespace creation
+      Reference ref = getReference();
+      Map<ContentKey, Content> contentMap = api.getContent().reference(ref).key(key).get();
```

Comment:
You can simplify this one in two (alternative) ways:
1. check if the returned map is empty (there can only be one element)
2. use `api.getContent().reference(ref).getSingle(key)` - however this one throws a NessieNotFoundException, if the content does not exist or if the ref does not exist - so it's probably more code (multiple catch-clauses)

Reply:
Option 1 doesn't allow to distinguish when the existing content is a namespace or something else.

Option 2 : it bothers me a bit that in the nominal case (content doesn't exist), we must expect to catch an exception. 

All in all I think this is fine, wdyt?